packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -78,11 +78,14 @@ export class SCWSigner implements Signer {
     return instance;
   }
 
-  async handshake() {
+  async handshake(args: RequestArguments) {
     const handshakeMessage = await this.createRequestMessage({
       handshake: {
-        method: 'eth_requestAccounts',
-        params: this.metadata,
+        method: args.method,
+        params: {
+          ...this.metadata,
```

Comment:
for backward compatibility.
can we make this cleaner though?

Reply:
`Object.assign({}, this.metadata, args.params ?? {})`